todoapp/TODOapp/app/src/main/java/com/nearsoft/androidschool/todoapp/activities/detail/DetailActivity.java

```diff
@@ -5,50 +5,163 @@
+            @Override
+            public void onClick(View view) {
+                try {
+                    saveData();
+                    Snackbar.make(editFab, "Data Saved (Not saving in reality)", Snackbar.LENGTH_SHORT).show();
+                    buttonViewConfig(false);
+                } catch (AssertionError er) {
+                    Snackbar.make(editFab, R.string.set_date_or_change_switch_message, Snackbar.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+    private void setEditListener() {
         editFab.setOnClickListener(new View.OnClickListener() {
             @Override
+            public void onClick(View view) {
+                buttonViewConfig(true);
+            }
+        });
+    }
+
+    private void setDateListener() {
+        dateCardView.setOnClickListener(new View.OnClickListener() {
+            @Override
             public void onClick(View v) {
-                Snackbar.make(editFab, "should open the \"edit\" activity", Snackbar.LENGTH_SHORT).show();
+                showDatePicker();
             }
         });
+    }
 
-        titleTextView.setText(todo.getTitle());
-        notesTextView.setText(todo.getNotes());
-        dateTextView.setText(todo.getDate());
+    private void showDatePicker() {
+        DatePickerFragment dialog = new DatePickerFragment();
+        dialog.attachListener(new android.app.DatePickerDialog.OnDateSetListener() {
+            @Override
+            public void onDateSet(DatePicker datePicker, int year, int month, int dayOfMonth) {
+                Date dateTime = new Date(year, month + 1, dayOfMonth, 0, 0);
+                updateDate(dateTime);
+            }
+        });
+        dialog.show(this.getSupportFragmentManager(), "datePicker");
     }
 
-    private ToDoContent getTodo() {
+    private void setSwitchListener() {
+        dateSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                dateCardView.setVisibility(isChecked ? View.VISIBLE : View.INVISIBLE);
+            }
+        });
+    }
+
+    private void buttonViewConfig(boolean isEditClicking) {
+        titleEditTextView.setEnabled(isEditClicking);
+        notesEditTextView.setEnabled(isEditClicking);
+        dateCardView.setEnabled(isEditClicking);
+        dateSwitch.setEnabled(isEditClicking);
+        editFab.setVisibility(isEditClicking ? View.INVISIBLE : View.VISIBLE);
+        saveFab.setVisibility(isEditClicking ? View.VISIBLE : View.INVISIBLE);
+    }
+
+    private void saveData() throws AssertionError {
+//        TODO:this method will make the saving
+        String selectedDate = dateTextView.getText().toString();
+        if (selectedDate.equals(getText(R.string.date))) {
+            throw new AssertionError(dateCardView.getClass().getSimpleName() + getString(R.string.set_date_or_change_switch_message));
```

Comment:
I do not understand this error, we throw an `AssertionError` if the selectedDate is the same as the date that was selected before?, why?, just ignore it, if it is the same, we dont update anything.